genetic_algorithms/algorithm_wrappers/algorithm_wrapper.py

```diff
@@ -0,0 +1,25 @@
+from abc import abstractmethod
+from genetic_algorithms.models.algorithm import Algorithm
+from genetic_algorithms.models.next_state_provider import NextStateProvider
+from genetic_algorithms.problems.base.state import State
+from genetic_algorithms.problems.base.model import Model
+
+
+class AlgorithmWrapper(NextStateProvider):
```

Comment:
To nowa abstrakcja która powinna pozwolić nam na takie rzeczy jak dodawanie wizualizacji oraz benchnarków.